rfcs/20221124-monorepo.md

```diff
@@ -0,0 +1,446 @@
+```
+
+Move everything into it except the following directories and files:
+
+| File / directory        | Reason                                 |
+| ----------------------- | -------------------------------------- |
+| .git/                   | For obvious reasons                    |
+| .github/                | Must be in the root                    |
+| .npmrc                  | For `pnpm` workspaces                  |
+| .pnpmfile.cjs           | For `pnpm` workspaces                  |
+| .editorconfig           | Useful across the monorepo             |
+| .eslintignore           | Read by ESLint running in pre-commit   |
+| .eslintrc.js            | Read by ESLint running in pre-commit   |
+| .prettierignore         | Read by Prettier running in pre-commit |
+| prettier.config.js      | Read by Prettier running in pre-commit |
+| .pre-commit-config.yaml | Read by pre-commit                     |
+| LICENSE                 | Root-level docs                        |
+| README.md               | Root-level docs                        |
+| CODE_OF_CONDUCT.md      | Root-level docs                        |
+| CONTRIBUTING.md         | Root-level docs                        |
+
+```bash
+$ git add . # Stage all renamed files
+$ git commit -m "Nest code under \`frontend/\`"
+```
+
+Switch to the monorepo.
+
+```bash
+$ git pull frontend main --allow-unrelated-histories
+```
+
+Since we fixed these files when merging the API, almost all these conflicts are
+redundant.
+
+| Expected conflicts      | Resolution                      |
+| ----------------------- | ------------------------------- |
+| .github/CODEOWNERS      | Use existing                    |
+| .prettierignore         | Use existing                    |
+| .pre-commit-config.yaml | Uncomment the commented regions |
+| .gitignore              | Use existing                    |
+| prettier.config.js      | Use existing                    |
+| README.md               | Use existing                    |
+| CONTRIBUTING.md         | Use existing                    |
+
+#### Housekeeping
+
+1. Create "stack: \*" labels to help with issue and PR management.
+   Spoiler/foreshadowing: these labels will be used for more things later.
+
+1. Migrate issues from `WordPress/openverse-frontend` and
+   `WordPress/openverse-api`. We can directly transfer the issues, retaining all
+   their comments. Apply the "stack: frontend" / "stack: backend" label to them
+   after moving.
+
+```bash
+# Substitute repo with WordPress/openverse-frontend and WordPress/openverse-api
+$ gh api \
+    -X GET \
+    repos/WordPress/<repo>/issues \
+    -f pulls=false \
+    -f state=all \
+    --jq '.[].number' \
+    --paginate \
+  | xargs \
+    -n 1 \
+    -I num \
+  gh issue transfer \
+    num \
+    WordPress/openverse \
+    -R WordPress/<repo>
+```
+
+With this done, we can archive the API and frontend repo. An optional notice may
+be added to the `README.md` files for clarification before archiving.
```

Comment:
```suggestion
With this done, we can archive the API and frontend repo. The following notice will
be added to the `README.md` files for clarification before archiving.
```

```md
> **Note**
>
> # This repository has moved to [WordPress/openverse](https://github.com/wordpress/openverse) as part of a monorepo.
```

Reply:
I updated your comment to ensure that merging it wouldn't break linting rules.